lms/services/course.py

```diff
@@ -299,6 +300,8 @@ def upsert_course(
             type_=Grouping.Type.COURSE,
             copied_from=copied_from,
         )[0]
+        # upsert_groupings returns Groupings of type `type_`
+        course = cast(Course, course)
```

Comment:
I'm sure we could make the type of upsert_groupings better. But it's not as easy as using `Grouping.Type.COURSE`.

Let's just tell mypy the right type here.